src/ts/application.ts

```diff
@@ -0,0 +1,416 @@
+import { ApplicationSettings } from "./profile/application_settings";
+import { SavegameManager } from "./savegame/savegame_manager";
+import { AboutState } from "./states/about";
+import { ChangelogState } from "./states/changelog";
+import { InGameState } from "./states/ingame";
+import { KeybindingsState } from "./states/keybindings";
+import { MainMenuState } from "./states/main_menu";
+import { MobileWarningState } from "./states/mobile_warning";
+import { PreloadState } from "./states/preload";
+import { SettingsState } from "./states/settings";
+import { ShapezGameAnalytics } from "./platform/browser/game_analytics";
+import { RestrictionManager } from "./core/restriction_manager";
+import { PuzzleMenuState } from "./states/puzzle_menu";
+import { ClientAPI } from "./platform/api";
+import { LoginState } from "./states/login";
+import { WegameSplashState } from "./states/wegame_splash";
+import { MODS } from "./mods/modloader";
+import { MOD_SIGNALS } from "./mods/mod_signals";
+import { ModsState } from "./states/mods";
+
+export type AchievementProviderInterface =
+    import("./platform/achievement_provider").AchievementProviderInterface;
+
+export type SoundInterface = import("./platform/sound").SoundInterface;
+
+export type StorageInterface = import("./platform/storage").StorageInterface;
+
+const logger = createLogger("application");
+
+// Set the name of the hidden property and the change event for visibility
+let pageHiddenPropName, pageVisibilityEventName;
+if (typeof document.hidden !== "undefined") {
+    // Opera 12.10 and Firefox 18 and later support
+    pageHiddenPropName = "hidden";
+    pageVisibilityEventName = "visibilitychange";
+    // @ts-ignore
+} else if (typeof document.msHidden !== "undefined") {
+    pageHiddenPropName = "msHidden";
+    pageVisibilityEventName = "msvisibilitychange";
+    // @ts-ignore
+} else if (typeof document.webkitHidden !== "undefined") {
+    pageHiddenPropName = "webkitHidden";
+    pageVisibilityEventName = "webkitvisibilitychange";
+}
+
+export class Application {
+    public unloaded: boolean;
+    public settings: ApplicationSettings;
+    public ticker: AnimationFrame;
+    public stateMgr: StateManager;
+    public savegameMgr: SavegameManager;
+    public inputMgr: InputDistributor;
+    public backgroundResourceLoader: BackgroundResourcesLoader;
+    public clientApi: ClientAPI;
+    public restrictionMgr: RestrictionManager;
+    public storage: StorageInterface;
+    public sound: any; // @Bagel03
+    public platformWrapper: PlatformWrapperInterface;
+    public achievementProvider: AchievementProviderInterface;
+    public adProvider: AdProviderInterface;
+    public analytics: AnalyticsInterface;
+    public gameAnalytics: ShapezGameAnalytics;
+    public focused: boolean;
+    public pageVisible: boolean;
+    public applicationPaused: boolean;
+    public trackedIsRenderable: TrackedState;
+    public trackedIsPlaying: TrackedState;
```

Comment:
These properties can by private as they are not used outside the application class